unit_tests/unit/provisioner/fake_azure_service.py

```
@@ -454,10 +455,125 @@ def __init__(self, path) -> None:
+    recover_after_polls: int | None = None
+
+
 class FakeVirtualMachines:
+    _scripts: Dict[str, Dict[str, Any]] = {}
+
     def __init__(self, path: Path) -> None:
         self.path = path
 
+    @classmethod
+    def set_provision_script(
+        cls, vm_name: str, behaviors: List[VMCreateBehavior] = None, default: VMCreateBehavior = None
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)